core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Util.java

```diff
@@ -186,7 +185,12 @@ private static OAuthTokenResponse refreshToken(
       }
 
       return fetchToken(
-          client, Map.of(), config.credential(), config.scope(), config.oauth2ServerUri());
+          client,
+          Map.of(),
+          config.credential(),
+          config.scope(),
+          config.oauth2ServerUri(),
+          optionalOAuthParams);
```

Comment:
The `fetchToken` call now forwards `optionalOAuthParams` (previous code didn’t). Is this a functional fix?